app/uk/gov/hmrc/helptosavestub/controllers/NSIGetAccountBehaviour.scala

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavestub.controllers
+
+import java.time.LocalDate
+
+import cats.instances.string._
+import cats.syntax.eq._
+import play.api.libs.json.{Format, Json}
+import ai.x.play.json.Jsonx
+import uk.gov.hmrc.helptosavestub.models.NSIErrorResponse
+
+object NSIGetAccountBehaviour {
+
+  def getAccountByNino(nino: String, correlationId: Option[String]): Either[NSIErrorResponse, NSIGetAccountByNinoResponse] =
```

Comment:
scalastyle warns about the cyclomatic complexity of this - perhaps converting it to a match statement would avoid the warning, i.e.
```scala
nino match {
}
```
if that doesn't work I'm happy for it to be left as is

Reply:
We get the same warning with it being a match statement, but its less lines of code being a match statement.